src/providers/http_provider.py

```diff
@@ -46,31 +45,35 @@ def _get(self, url: str, params: Optional[dict] = None) -> Tuple[dict | list, di
         """
         Returns (data, meta)
         """
-        with self.PROMETHEUS_HISTOGRAM.time():
-            response = self.session.get(
-                urljoin(self.host, url),
-                params=params,
-                timeout=self.REQUEST_TIMEOUT,
-            )
+        request_name = self._url_to_request_name_label(url)
+        with self.PROMETHEUS_HISTOGRAM.time() as t:
+            try:
+                response = self.session.get(
+                    urljoin(self.host, url),
+                    params=params,
+                    timeout=self.REQUEST_TIMEOUT,
+                )
+                if response.status_code != HTTPStatus.OK:
+                    msg = f'Response [{response.status_code}] with text: "{str(response.text)}" returned.'
+                    logger.debug({'msg': msg})
+                    raise NotOkResponse(msg, status=response.status_code, text=response.text)
 
-        if response.status_code != HTTPStatus.OK:
-            msg = f'Response [{response.status_code}] with text: "{str(response.text)}" returned.'
-            logger.debug({'msg': msg})
-            raise NotOkResponse(msg, status=response.status_code, text=response.text)
-
-        try:
-            json_response = response.json()
-            data = json_response['data']
-            del json_response['data']
-        except (KeyError, JSONDecodeError) as error:
-            msg = f'Response [{response.status_code}] with text: "{str(response.text)}" returned.'
-            logger.debug({'msg': msg})
-            raise error from error
-        finally:
-            self.PROMETHEUS_COUNTER.labels(
-                method='get',
-                code=response.status_code,
-                domain=urlparse(self.host).netloc,
-            ).inc()
+                json_response = response.json()
+                data = json_response['data']
+                del json_response['data']
+            except (KeyError, JSONDecodeError) as error:
+                msg = f'Response [{response.status_code}] with text: "{str(response.text)}" returned.'
+                logger.debug({'msg': msg})
+                raise error from error
+            finally:
+                t.labels(
+                    name=request_name,
+                    code=response.status_code,
+                    domain=urlparse(self.host).netloc,
+                )
 
         return data, json_response
+
+    @abstractmethod
+    def _url_to_request_name_label(self, url: str) -> str:
```

Comment:
Maybe just pass original url to _get function
This method wont work for finalized and head

Reply:
It's expected behaviour

Reply:
We can pass to `_get` two separated args - endpoint and params, but I'm not sure that it's a good decision

Reply:
fixed